controller/api/destination/server.go

```diff
@@ -467,88 +442,37 @@ func (s *server) subscribeToServiceWithoutContext(
 //
 // This function does not return until the stream is closed.
 func (s *server) subscribeToEndpointProfile(
-	address *watcher.Address,
+	service *watcher.ServiceID,
+	hostname,
+	ip string,
 	port uint32,
 	stream pb.Destination_GetProfileServer,
 ) error {
-	log := s.log
-	if address.Pod != nil {
-		log = log.WithField("ns", address.Pod.Namespace).WithField("pod", address.Pod.Name)
-	} else {
-		log = log.WithField("ip", address.IP)
-	}
+	translator := newEndpointProfileTranslator(
+		s.enableH2Upgrade,
+		s.controllerNS,
+		s.identityTrustDomain,
+		s.defaultOpaquePorts,
+		stream,
+		s.k8sAPI,
+		s.metadataAPI,
+	)
 
-	opaquePorts := getAnnotatedOpaquePorts(address.Pod, s.defaultOpaquePorts)
-	var endpoint *pb.WeightedAddr
-	endpoint, err := s.createEndpoint(*address, opaquePorts)
+	var err error
+	ip, err = s.pods.Subscribe(service, hostname, ip, port, translator)
```

Comment:
I pushed that change, but the linter is rejecting it because `ip` is already defined (as a function parameter). I'm gonna revert the change, even though my local linter version isn't complaning :man_shrugging:

Reply:
Ah :( yeah I thought it might've been like this because of some linter complaints. fwiw I have also tried it locally before suggesting the change and it seemed to work. Sounds good to me!